### PR TITLE
fix(DT): correct DT set between 10pm and noon

### PR DESCRIPTION
**Pull request by sc979**
## Description

- Fix the downtime set between 10pm and noon.
- Remove the modification on the datepicker start value when setting an end value before it.
- Add a popup to display warning for the customer to understand why and how the downtime end time has been changed.
image

**Fixes** # (MON-3935)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [x] 19.04.x
- [x] 19.10.x
- [x] 20.04.x (master)

## Checklist

- [x] I followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/include/common/javascript/datepicker/localizedDatepicker.js [hunk 1]
```diff
@@ -132,58 +130,100 @@ function setUserFormat() {
...
+function checkEndTime() {
+    let startTime = $(".timepicker").first().val()
+    let start = moment($('[name="alternativeDateStart"]').val()
+        + ' ' + startTime, "MM/DD/YYYY HH:mm");
+
+    let endTime = $(".timepicker").last().val();
+    let end = moment($('[name="alternativeDateEnd"]').val()
+        + ' ' + endTime, "MM/DD/YYYY HH:mm");
+
+    if (start.isAfter(end) || start.isSame(end)) {
```
Comment (kduret) [inline www/include/common/javascript/datepicker/localizedDatepicker.js:221]: you could use `isSameOrAfter` : https://momentjscom.readthedocs.io/en/latest/moment/05-query/05-is-same-or-after/

File: www/include/common/javascript/datepicker/localizedDatepicker.js [hunk 2]
```diff
@@ -132,58 +130,100 @@ function setUserFormat() {
...
+ */
+function updateEndTime() {
     let start = moment($('[name="alternativeDateStart"]').val()
-        + ' ' +  $(".timepicker").first().val(), "MM/DD/YYYY HH:mm");
+        + ' ' + $(".timepicker").first().val(), "MM/DD/YYYY HH:mm");
     let end = moment($('[name="alternativeDateEnd"]').val()
-        + ' ' +  $(".timepicker").last().val(), "MM/DD/YYYY HH:mm");
+        + ' ' + $(".timepicker").last().val(), "MM/DD/YYYY HH:mm");
 
     if (start.isAfter(end) || start.isSame(end)) {
```
Comment (kduret) [inline www/include/common/javascript/datepicker/localizedDatepicker.js:199]: you could use `isSameOrAfter` : https://momentjscom.readthedocs.io/en/latest/moment/05-query/05-is-same-or-after/

File: www/include/common/javascript/datepicker/localizedDatepicker.js [hunk 3]
```diff
@@ -132,58 +130,100 @@ function setUserFormat() {
...
+    let start = moment($('[name="alternativeDateStart"]').val()
+        + ' ' + startTime, "MM/DD/YYYY HH:mm");
+
+    let endTime = $(".timepicker").last().val();
+    let end = moment($('[name="alternativeDateEnd"]').val()
+        + ' ' + endTime, "MM/DD/YYYY HH:mm");
+
+    if (start.isAfter(end) || start.isSame(end)) {
+        // Display a warning to the user.
+        alert("The downtime end time - " + endTime + ",\nis unconsistent with the start time - " + startTime +
```
Comment (kduret) [inline www/include/common/javascript/datepicker/localizedDatepicker.js:223]:
```suggestion
        alert("The downtime end time - " + endTime + ",\nis not consistent with the start time - " + startTime +
```

File: www/include/common/javascript/datepicker/localizedDatepicker.js (hunk 1 above)
Comment (sc979) [reply www/include/common/javascript/datepicker/localizedDatepicker.js]: done

File: www/include/monitoring/external_cmd/popup/templates/massive_downtime.ihtml [hunk 4]
```diff
@@ -72,6 +72,6 @@
     jQuery(".timepicker").timepicker();
     initDatepicker();
     turnOnEvents();
-    updateEndTime();
+	updateDateAndTime();
```
Comment (kduret) [inline www/include/monitoring/external_cmd/popup/templates/massive_downtime.ihtml:75]:
```suggestion
    updateDateAndTime();
```

**Pull request merged**